Library/Homebrew/sandbox.rb

```diff
@@ -68,6 +69,11 @@ def allow_write_cellar(formula)
     allow_write_path formula.var
   end
 
+  # Xcode projects expect access to certain cache/archive dirs.
+  def allow_write_xcode
+    allow_write_path "/Users/#{ENV["USER"]}/Library/Developer/Xcode/DerivedData/"
```

Comment:
I think the pointer as to where the Xcode directories are is provided by Xcode itself sadly. It looks possible to [overwrite](https://developer.apple.com/library/ios/recipes/xcode_help-locations_preferences/DerivedData/DerivedData.html) either in Xcode or via the command-line, but whether or not we want to enforce that on every project right away is probably premature.